mitiq/zne/tests/test_zne.py

```diff
@@ -492,6 +493,34 @@ def test_execute_with_zne_with_supported_circuits(circuit_type):
     assert abs(unmitigated - expected) > abs(zne_value - expected)
 
 
+@pytest.mark.parametrize("circuit_type", SUPPORTED_PROGRAM_TYPES.keys())
+def test_layerwise_execute_with_zne_with_supported_circuits(circuit_type):
+    # Define a circuit equivalent to the identity
+    qreg = cirq.LineQubit.range(2)
+    cirq_circuit = cirq.Circuit(
+        cirq.H.on_each(qreg),
+        cirq.CNOT(*qreg),
+        cirq.CNOT(*qreg),
+        cirq.H.on_each(qreg),
+    )
+    # Convert to one of the supported program types
+    circuit = convert_from_mitiq(cirq_circuit, circuit_type)
+    expected = generic_executor(circuit, noise_level=0.0)
+    unmitigated = generic_executor(circuit)
+    # Use odd scale factors for deterministic results
+    fac = RichardsonFactory([1, 3, 5])
+    # Layerwise-fold
+    layer_to_fold = 0
+    fold_layer_func = get_layer_folding(layer_to_fold)
+
+    zne_value = execute_with_zne(
+        circuit, generic_executor, factory=fac, scale_noise=fold_layer_func
+    )
+
+    # Test zero noise limit is better than unmitigated expectation value
+    assert abs(unmitigated - expected) > abs(zne_value - expected)
```

Comment:
I was a little worried this might not be deterministic enough, but I ran the test 2k times and it didn't fail, so that's good enough for me!

Reply:
Haha, appreciate the check!